contracts/fate.md

```diff
@@ -0,0 +1,648 @@
+| 0x2b | 'MAP_EMPTY' | Arg0 | Arg0 := #{}. |
+| 0x2c | 'MAP_LOOKUP' | Arg0 Arg1 Arg2 | Arg0 := lookup key Arg2 in map Arg1. |
+| 0x2d | 'MAP_LOOKUPD' | Arg0 Arg1 Arg2 Arg3 | Arg0 := lookup key Arg2 in map Arg1 if key exists in map otherwise Arg0 := Arg3. |
+| 0x2e | 'MAP_UPDATE' | Arg0 Arg1 Arg2 Arg3 | Arg0 := update key Arg2 in map Arg1 with value Arg3. |
+| 0x2f | 'MAP_DELETE' | Arg0 Arg1 Arg2 | Arg0 := delete key Arg2 from map Arg1. |
+| 0x30 | 'MAP_MEMBER' | Arg0 Arg1 Arg2 | Arg0 := true if key Arg2 is in map Arg1. |
+| 0x31 | 'MAP_FROM_LIST' | Arg0 Arg1 | Arg0 := make a map from (key, value) list in Arg1. |
+| 0x32 | 'NIL' | Arg0 | Arg0 := []. |
+| 0x33 | 'IS_NIL' | Arg0 Arg1 | Arg0 := true if Arg1 == []. |
+| 0x34 | 'CONS' | Arg0 Arg1 Arg2 | Arg0 := [Arg1|Arg2]. |
+| 0x35 | 'HD' | Arg0 Arg1 | Arg0 := head of list Arg1. |
+| 0x36 | 'TL' | Arg0 Arg1 | Arg0 := tail of list Arg1. |
+| 0x37 | 'LENGTH' | Arg0 Arg1 | Arg0 := length of list Arg1. |
+| 0x38 | 'STR_EQ' | Arg0 Arg1 Arg2 | Arg0 := true iff the strings Arg1 and Arg2 are the same. |
+| 0x39 | 'STR_JOIN' | Arg0 Arg1 Arg2 | Arg0 := string Arg1 followed by string Arg2. |
+| 0x40 | 'INT_TO_STR' | Arg0 Arg1 | Arg0 := turn integer Arg1 into a string. |
+| 0x41 | 'ADDR_TO_STR' | Arg0 Arg1 | Arg0 := turn address Arg1 into a string. |
+| 0x42 | 'STR_REVERSE' | Arg0 Arg1 | Arg0 := the reverse of string Arg1. |
+| 0x43 | 'INT_TO_ADDR' | Arg0 Arg1 | Arg0 := turn integer Arg1 into an address. |
+| 0x44 | 'VARIANT' | Arg0 Arg1 Arg2 Arg3 | Arg0 := create a variant of size Arg1 with the tag Arg2 (Arg2 < Arg1) and take Arg3 elements from the stack. |
+| 0x45 | 'VARIANT_TEST' | Arg0 Arg1 Arg2 | Arg0 := true if variant Arg1 has the tag Arg2. |
+| 0x46 | 'VARIANT_ELEMENT' | Arg0 Arg1 Arg2 | Arg0 := element number Arg2 from variant Arg1. |
+| 0x47 | 'BITS_NONEA' |  | accumulator := empty bitmap. |
+| 0x48 | 'BITS_NONE' | Arg0 | Arg0 := empty bitmap. |
+| 0x49 | 'BITS_ALLA' |  | accumulator := full bitmap. |
+| 0x50 | 'BITS_ALL' | Arg0 | Arg0 := full bitmap. |
+| 0x51 | 'BITS_ALL_N' | Arg0 Arg1 | Arg0 := bitmap with Arg1 bits set. |
+| 0x52 | 'BITS_SET' | Arg0 Arg1 Arg2 | Arg0 := set bit Arg2 of bitmap Arg1. |
+| 0x53 | 'BITS_CLEAR' | Arg0 Arg1 Arg2 | Arg0 := clear bit Arg2 of bitmap Arg1. |
+| 0x54 | 'BITS_TEST' | Arg0 Arg1 Arg2 | Arg0 := true if bit Arg2 of bitmap Arg1 is set. |
+| 0x55 | 'BITS_SUM' | Arg0 Arg1 | Arg0 := sum of set bits in bitmap Arg1. Exception if infinit bitmap. |
+| 0x56 | 'BITS_OR' | Arg0 Arg1 Arg2 | Arg0 := Arg1 v Arg2. |
+| 0x57 | 'BITS_AND' | Arg0 Arg1 Arg2 | Arg0 := Arg1 ^ Arg2. |
+| 0x58 | 'BITS_DIFF' | Arg0 Arg1 Arg2 | Arg0 := Arg1 - Arg2. |
+| 0x59 | 'ADDRESS' | Arg0 | Arg0 := The current contract address. |
+| 0x5a | 'BALANCE' | Arg0 | Arg0 := The current contract address. |
+| 0x5b | 'ORIGIN' | Arg0 | Arg0 := Address of contract called by the call transaction. |
+| 0x5c | 'CALLER' | Arg0 | Arg0 := The address that signed the call transaction. |
```

Comment:
As discussed with @happi and @gorillainduction , may need review.

Reply:
One minor thing, but with generalised accounts, the »signed« could be no longer accurate, so it might be good to start thinking of a more general notion of »authorised«.